src/main/java/success/planfit/service/UserService.java

```diff
@@ -22,4 +25,46 @@ public User findById(Long id) {
         return userRepository.findById(id)
                 .orElseThrow(() -> new IllegalArgumentException("해당 id로 회원이 조회되지 않음."));
     }
+    // 회원 정보 조회 (기본키로 조회)
+    public UserDto getUserInfo(Long userId) {  // 회원 기본키 (userId)를 파라미터로 받음
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("회원 정보 조회 실패"));
+
+        return UserDto.from(user);  // UserDto로 변환하여 반환
+    }
+
+    // 회원 정보 수정
+    @Transactional
+    public void updateUserInfo(Long userId, UserDto userDto) {
+        User user = userRepository.findById(userId)  // ID로 조회
+                .orElseThrow(() -> new RuntimeException("회원 정보 조회 실패"));
+
+        if (userDto.getProfilePhoto() != null) {
+            user.setProfilePhoto(PhotoProvider.decode(userDto.getProfilePhoto()));
+        }
+        if (userDto.getBirthOfDate() != null) {
+            user.setBirthOfDate(userDto.getBirthOfDate());
+        }
+        if (userDto.getName() != null) {
+            user.setName(userDto.getName());
+        }
+        if (userDto.getPhoneNumber() != null) {
+            user.setPhoneNumber(userDto.getPhoneNumber());
+        }
+        if (userDto.getIdentity() != null) {
+            user.setIdentity(userDto.getIdentity());
+        }
+        if (userDto.getEmail() != null) {
+            user.setEmail(userDto.getEmail());
+        }
+        if (userDto.getPassword() != null) {
+            if (user instanceof PlanfitUser) {
+                PlanfitUser planfitUser = (PlanfitUser) user;
+                planfitUser.setPassword(userDto.getPassword());
+            }
+
+        }
```

Comment:
```suggestion
        if (userDto.getPassword() != null && user instanceof PlanfitUser) {
            PlanfitUser planfitUser = (PlanfitUser) user;
            planfitUser.setPassword(userDto.getPassword());
        }
```
조건문을 위와 같이 합치는 것이 가독성 측면에서 더 좋지 않을까요?
그리고 `PlanfitUser`의 경우 `password` 뿐 아니라 `loginId` 정보도 가지고 있는데, `loginId`를 수정하는 코드는 보이지 않네요.
의도된 사항인가요?

Reply:
아이디를 수정하는 경우는 거의 없다고 생각해서 password만 수정하게끔 했습니다.